### PR TITLE
fix: correct the factor of 2 in the Tweedie unit deviance

### PR DESCRIPTION
**Pull request by nasaul**
## Problem

The unit Tweedie deviance multiplies the whole bracket by 2, but `tweedie_deviance` only doubled the first term in the `power == 2` and general-power branches:

```python
2 * ( y^(2-p) / ((1-p)(2-p)) )   # doubled
- ( y * mu^(1-p) / (1-p) )       # not doubled
+ ( mu^(2-p) / (2-p) )           # not doubled
```

So every power other than 0 and 1 was wrong, and `1 < power < 2` returned negative values — impossible for a deviance. `test_loss` only covered powers 0, 1 and 2, and its own `power == 2` reference repeated the same error, so nothing caught it.

## Fix

`tweedie_deviance` now builds its expression in a single `gen_expr` with named `target`/`pred` locals instead of four separate closures:

- gamma (`power == 2`): the ratio form `2 * (r - log(r) - 1)`, one `log` call instead of two.
- general power: all three terms grouped under the factor of 2.
- Poisson (`power == 1`): takes `y * log(y / mu) -> 0` at `y == 0` directly, rather than branching to a separate closed form for that row.
- `power == 0` stays its own case so squared error keeps working for negative targets (the general branch clips at 0).

The gamma case in the docs formula had the same misplaced factor and is corrected to match.

## Verification

- New `test_tweedie_deviance_powers` sweeps powers 0, 1, 1.2, 1.5, 1.8, 2, 2.5, 3 on both pandas and polars against a numpy reference, and asserts the two invariants the old code broke: the deviance is non-negative, and it is 0 for a perfect forecast. 12 of its 16 cases fail against the old implementation.
- Cross-checked against `sklearn.metrics.mean_tweedie_deviance` for those powers (max abs diff < 1e-10), including targets of 0 at powers 0 and 1, and negative targets at power 0.
- `tests/test_losses.py`: 120 passed. Full suite: 1744 passed, with `test_distributed_evaluate` failing identically on clean `main` (local Spark/py4j).

### CODE REVIEW

File: utilsforecast/losses.py [hunk 1]
```diff
@@ -1206,49 +1206,30 @@ def tweedie_deviance(
...
-                + (nw.col(model) ** (2 - power) / (2 - power))
-            ).alias(model)
+    def gen_expr(model):
+        pred = nw.col(model)
+        if power == 0:
+            deviance = (target - pred) ** 2
+        elif power == 1:
+            # y * log(y / mu) is taken as 0 at y == 0, its limit there
+            deviance = 2 * (
+                nw.when(target > 0).then(target * (target / pred).log()).otherwise(0.0)
```
Comment (ngupta23) [inline utilsforecast/losses.py:1218]: Low. The guard is `target > 0`, but the `y·log(y/µ) → 0` substitution is only valid as the limit at `y == 0` (as the comment above says). Negative targets reach this branch too, because the domain check on line 1200 only fires for `power >= 2`.

  ```python
  tweedie_deviance(
      pd.DataFrame({"unique_id": "a", "y": [-10.0, -5.0], "model": [1.0, 1.0]}),
      ["model"], power=1,
  )
  # this branch: 17.0   |   main: nan
  ```

So out-of-domain input that used to announce itself as NaN now returns a plausible-looking finite loss, on both pandas and polars. Either narrow the guard to `target == 0` (keeping the NaN signal) or extend the validation on line 1200 to reject `y < 0` when `power >= 1`.

File: utilsforecast/losses.py [hunk 2]
```diff
@@ -1206,49 +1206,30 @@ def tweedie_deviance(
...
+                - target
+                + pred
+            )
+        elif power == 2:
+            ratio = target / pred
+            deviance = 2 * (ratio - ratio.log() - 1)
+        else:
+            # every term of the unit deviance carries the factor of 2
+            deviance = 2 * (
+                target.clip(0) ** (2 - power) / ((1 - power) * (2 - power))
```
Comment (ngupta23) [inline utilsforecast/losses.py:1228]: Low. Only the first term is clipped; the third term on line 1230 uses the raw `target`. For `1 < p < 2` a negative actual therefore yields a negative deviance — `y = -10, µ = 1, p = 1.5` → `-36.0, p = 1.2` → `-97.5`.

This is pre-existing (`main` gives `-18.0` for the same input), but the PR states non-negativity as the invariant and this line is in the diff. A `power >= 1` target validation would be more honest than the asymmetric clip.

File: tests/test_losses.py [hunk 3]
```diff
@@ -635,3 +636,47 @@ def frames(id_col):
...
+    np.testing.assert_allclose(res["model"].iloc[0], expected)
+    # a deviance vanishes for a perfect forecast
+    np.testing.assert_allclose(res["perfect"].iloc[0], 0.0, atol=1e-12)
+
+
+@pytest.mark.parametrize("power", TWEEDIE_POWERS)
+def test_tweedie_deviance_vs_sklearn(power):
+    sklearn_metrics = pytest.importorskip("sklearn.metrics")
+    y, pred = tweedie_sample()
+    if power <= 1:
```
Comment (ngupta23) [inline tests/test_losses.py:674]: Low (coverage). `y == 0` is in the domain for `1 < p < 2` as well, and that is exactly the `clip(0)` general branch this PR rewrote — which the new tests otherwise exercise only with `y >= 0.5`. I checked the branch is correct at `y == 0` (matches sklearn for `p = 1.2/1.5/1.8`), so no live bug, but `if power < 2`: here would pin the behavior the rewrite depends on.

**Review by ngupta23 — APPROVED** [review]
